backend/ethereum/channel/adjudicator_test.go

```diff
@@ -33,6 +34,128 @@ import (
 	pkgtest "perun.network/go-perun/pkg/test"
 )
 
+// TestAdjudicator tests the adjudicator.
+func TestAdjudicator(t *testing.T) {
+	ctx, cancel := context.WithTimeout(context.Background(), defaultTestTimeout)
+	defer cancel()
+
+	rng := pkgtest.Prng(t)
+	numParts := 2 + rng.Intn(maxNumParts-2)
+	s := test.NewSetup(t, rng, numParts, blockInterval)
+	a := newAdjudicatorSetup(s)
+	channeltest.TestAdjudicator(ctx, t, a)
+}
+
+type adjudicatorSetup struct {
+	setup *test.Setup
+}
+
+func newAdjudicatorSetup(setup *test.Setup) *adjudicatorSetup {
+	return &adjudicatorSetup{
+		setup: setup,
+	}
+}
+
+type testAdjudicator struct {
+	*test.SimAdjudicator
+}
+
+// Progress is a no-op because app channels are not supported yet.
+func (a *testAdjudicator) Progress(ctx context.Context, req channel.ProgressReq) error {
+	return nil
+}
+
+func (a *adjudicatorSetup) Adjudicator() channel.Adjudicator {
+	return &testAdjudicator{a.setup.Adjs[0]}
+}
+
+func (a *adjudicatorSetup) NewRegisterReq(ctx context.Context, rng *rand.Rand) (*channel.AdjudicatorReq, []channel.SignedState) {
+	req := a.newAdjudicatorReq(ctx, rng, channeltest.WithIsFinal(false))
+	return req, nil
+}
+
+func (a *adjudicatorSetup) newAdjudicatorReq(ctx context.Context, rng *rand.Rand, opts ...channeltest.RandomOpt) *channel.AdjudicatorReq {
+	params, state := a.newRandomParamsAndState(rng, opts...)
+
+	// Fund the channel.
+	numParts := len(params.Parts)
+	errs := make(chan error, numParts)
+	for i, funder := range a.setup.Funders {
+		req := channel.NewFundingReq(params, state, channel.Index(i), state.Balances)
+		go func(funder channel.Funder, req channel.FundingReq) {
+			errs <- funder.Fund(ctx, req)
```

Comment:
You can use the error gatherer here.

Reply:
This comes back to the other discussion. I am aware of the error gatherer, but I find it cleaner to use standard go functionality here.